### ISSUE TITLE
Missing/invalid JSON causes an unhandled exception 500 error

### ISSUE THREAD

**Issue by jeffw-wherethebitsroam** [graphql-dotnet/server#580]
I set up a test graphql server using:

```
            services.AddSingleton<TestSchema>()
                .AddGraphQL()
                .AddSystemTextJson()
                .AddGraphTypes(typeof(TestSchema));
```

```
            app.UseEndpoints(endpoints =>
            {
                endpoints.MapGraphQL<TestSchema>();
            });
```

If the body of the graphql request is then invalid, I get a 500 response and an error like:

```
      An unhandled exception has occurred while executing the request.
      System.InvalidOperationException: Cannot execute request if no query is specified
         at GraphQL.DocumentExecuter.ExecuteAsync(ExecutionOptions options) in /_/src/GraphQL/Execution/DocumentExecuter.cs:line 66
```

Firstly, shouldn't the middleware return some sort of 400 error? This is clearly a client error.

Secondly, the `System.InvalidOperationException: Cannot execute request if no query is specified` completely obscures the actual error, making it very hard to debug misbehaving clients.

For example, we recently had a problem with a client sending a newline in their JSON, i.e.

```
curl -v https://localhost:5001/graphql -d '{"query": "{ hello\n }"}' -H 'application/json'
```

And the only error reported is `Cannot execute request if no query is specified` which is very misleading.

**Comment by alilengrai** [comment]
UP

**Comment by sungam3r** [comment]
Probably fixed by #493. I have not been engaged in this project for a long time. Closer to winter I plan to return to support.

**Comment by Shane32** [comment]
Should check again - may have already been solved by #581

**Comment by sungam3r** [comment]
Fixed in https://github.com/graphql-dotnet/server/releases/tag/5.1.0
